gslab_scons/tests/test_build_python.py

```diff
@@ -3,40 +3,124 @@
 import sys
 import os
 import shutil
+import mock
+import subprocess
 
 sys.path.append('../..')
-from gslab_scons import build_python
+import gslab_scons.builders.build_python as gs
 from gslab_scons._exception_classes import BadExtensionError
+from gslab_make.tests import nostderrout
 
-
-class testbuild_python(unittest.TestCase):
+class TestBuildPython(unittest.TestCase):
 
     def setUp(self):
-        if not os.path.exists('../build/'):
-            os.mkdir('../build/')
-
-    def test_default(self):
-        env = ''
-        build_python('../build/py.py', './input/python_test_script.py', env)
-        logfile_data = open('../build/sconscript.log', 'rU').read()
-        self.assertIn('Log created:', logfile_data)
-        if os.path.isfile('../build/sconscript.log'):
-            os.remove('../build/sconscript.log')
- 
+        if not os.path.exists('./build/'):
+            os.mkdir('./build/')
+
+    def test_log_creation(self):
+        '''
+        Test that build_python() creates a log in its
+        target's directory.
+        '''
+        gs.build_python(target = './build/py.py', 
+                       source = './input/python_test_script.py', 
+                       env    = '')
+        self.check_log('./build/sconscript.log')
+
+    def check_log(self, log_path = './build/sconscript.log'):
```

Comment:
Do we want to have a set of misc functions that we load for all unit tests?

At one extreme, it seems like we do a lot of the same tests across builders (bad extension, log creation, etc) where the only differences across these items are the build function, the source, and the target. Would it make sense/be possible to have a single `test_builders.py` script that performs the same tests across all of our builders?

Using something like http://stackoverflow.com/questions/14078357/python-how-can-idynamically-create-a-function-with-a-name-based-on-a-string-re

Let me know if this isn't clear.